packages/ozone/src/api/communication/updateTemplate.ts

```diff
@@ -26,14 +26,27 @@ export default function (server: Server, ctx: AppContext) {
       }
 
       const communicationTemplate = ctx.communicationTemplateService(db)
-      const updatedTemplate = await communicationTemplate.update(Number(id), {
-        ...template,
-        lastUpdatedBy: updatedBy,
-      })
+      try {
+        const updatedTemplate = await communicationTemplate.update(Number(id), {
+          ...template,
+          lastUpdatedBy: updatedBy,
+        })
 
-      return {
-        encoding: 'application/json',
-        body: communicationTemplate.view(updatedTemplate),
+        return {
+          encoding: 'application/json',
+          body: communicationTemplate.view(updatedTemplate),
+        }
+      } catch (err) {
+        if (
+          err?.['code'] === '23505' &&
+          err?.['constraint'] === 'communication_template_unique_name'
+        ) {
```

Comment:
I recognize this as a postgres-specific error code, but I could see this being confusing to a reader.  Perhaps it could be self-documented pretty well by simply pulling it out into a util.